persistence-api/src/main/java/org/apache/cassandra/stargate/db/DeleteEvent.java

```diff
@@ -0,0 +1,10 @@
+package org.apache.cassandra.stargate.db;
+
+import java.util.List;
+
+/** Represents a delete of one or multiple rows. */
+public interface DeleteEvent extends MutationEvent {
```

Comment:
I understand that this will be handling delete of the whole row, right?
What about the deletion of one field in a row? Will it be handled as the `RowUpdateEvent` with the cell having `null` value? (denoting clear/delete)?

Reply:
> Will it be handled as the RowUpdateEvent with the cell having null value? (denoting clear/delete)?

yes!

`DeleteEvent` can affect one or multiple rows depending on the amount of keys specified.

Reply:
Does it mean that it will be possible to get only PKs without CK for delete? Such a solution is problematic because the Kafka key should contain all PKs and Clustering Keys. 
If we omit any of those keys, sending null as any Kafka field value will mean that the delete event will be sent to a different Kafka partition than an update. This will break the time ordering because data for the same record will not land on the same Kafka partition. 
Debezium treats all PKs and Clustering Keys as required for that reason:
https://debezium.io/documentation/reference/connectors/cassandra.html#cassandra-change-events-value

Reply:
ok, so we should leave out partition delete (when CKs exist) and range deletes for now.

Reply:
Now that I think of it, I think the record key should only be composed by the partition keys (and not the CKs).

Reply:
This may be problematic - Kafka will have performance degradation because of the partition skew

Reply:
Having partition keys would be easier to reason about and it would equate partitioning on both Kafka and C*: skew on one side would mean skew on the other.

Reply:
agree, I see one more small disadvantage of not having clustering keys in the `kafka.key`: it will be a bit harder for clients of CDC to locate the record in db based only on the `kafka.key` but maybe that is not a problem. 
I'll try to rework and see how it will look like; so to sum up:
- for `RowUpdateEvent` the `getPartitionKeys` and `getClusteringKeys` will always return all columns. PKs will be in `kafka.key` whereas CKs in `kafka.value`
- for the `DeleteEvent` CKs may or may not be present. The value schema seems already refecting that.
The most important bit for the client to react accordingly will be (`op: u` or `op: d`)